pkgs/applications/networking/instant-messengers/zoom-us/default.nix

```diff
@@ -66,15 +74,27 @@ in mkDerivation {
       runHook postInstall
     '';
 
-  postInstall = (makeDesktopItem {
-    name = "zoom-us";
-    exec = "$out/bin/zoom-us %U";
-    icon = "$out/share/zoom-us/application-x-zoom.png";
-    desktopName = "Zoom";
-    genericName = "Video Conference";
-    categories = "Network;Application;";
-    mimeType = "x-scheme-handler/zoommtg;";
-  }).buildCommand + ''
+  postInstall = ''
+    mkdir -p $out/share/{applications,appdata,icons}
+
+    # Desktop File
+    cp ${desktopIntegration}/us.zoom.Zoom.desktop $out/share/applications
+    substituteInPlace $out/share/applications/us.zoom.Zoom.desktop \
+        --replace "Exec=zoom" "Exec=$out/bin/zoom-us"
```

Comment:
Note that this isn't strictly necessary. Although it also doesn't hurt. Eventually I would like an official policy on this, but right now we mix absolute Exec and PATH-based Exec's.

Reply:
We definitely need some docs on guidelines for freedesktop things, and I think that would be one of them.
PATH-based Exec's have been a problem in the past when it needs some executable that
would be in `glib.dev` for example.

Reply:
Absolute Exec entries can be a pain when updating, though, as desktop files in the user profile aren't picked up. Is there a way to force, e.g., GNOME 3 to rebuild its desktop file database?

Reply:
When you say
>  as desktop files in the user profile aren't picked up

Do you mean programs installed with `nix-env` don't get included or the entry in an application menu doesn't get updated?

Think that's a common issue #12757.

Reply:
> the entry in an application menu doesn't get updated?

This one, and thanks. I don't think we need to address this here, so what you have is fine.